src/hooks/job-queues/use-submit-job-queue.js

```diff
@@ -0,0 +1,36 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import history from '../../services/history'
+import formatError from '../../services/format-error'
+
+const createMutation = (name) => ({
+    resource: `scheduler/queues/${name}`,
+    type: 'create',
+    data: ({ queue }) => queue,
+})
+
+const useSubmitJobQueue = () => {
+    const engine = useDataEngine()
+    const submitJobQueue = (queue) => {
+        const mutation = createMutation(queue.name)
+        return engine
+            .mutate(mutation, { variables: { queue } })
+            .then(() => {
+                history.push('/')
+            })
```

Comment:
I was thinking that it might be good to first finish the edit form and the list and then take a look at cleaning up these kinds of patterns. There's a fair bit of cleanup to do I think, maybe getting the features out of the door first would be a good initial goal? Let me know what you think.

Reply:
For future reference, it's this issue: https://dhis2.atlassian.net/browse/DHIS2-15476. I'll work on that next.